### PR TITLE
added example JNB for xcube cmems

### PR DESCRIPTION
**Pull request by TejasMorbagal**
[Description of PR]
added example JNB for xcube-cmems

Checklist:

* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in `docs/source/*`
* [ ] Changes documented in `CHANGES.md`
* [ ] AppVeyor CI passes
* [ ] Test coverage remains or increases (target 100%)

### CODE REVIEW

**Review by pont-us — APPROVED** [review]

File: examples/notebooks/datastores/7_cmems_data_store.ipynb [hunk 1]
```diff
@@ -0,0 +1,2538 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "f1b35e6b-22c6-4977-a286-b26e1db65f66",
+   "metadata": {},
+   "source": [
+    "Please checkout the general introduction to xcube data stores in the Jupyter Notebook Getting Started before jumping into this notebook :)\n",
```
Comment (TonioF) [inline examples/notebooks/datastores/7_cmems_data_store.ipynb:8]: Can you put this in italics, please?

File: examples/notebooks/datastores/7_cmems_data_store.ipynb [hunk 2]
```diff
@@ -0,0 +1,2538 @@
...
+   "source": [
+    "get_data_store_params_schema('cmems')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "214307bc-7def-433f-876e-28540a6c2fe7",
+   "metadata": {},
+   "source": [
+    "#### set `CMEMS_USER` and `CMEMS_PASSWORD` as environment variables"
```
Comment (TonioF) [inline examples/notebooks/datastores/7_cmems_data_store.ipynb:151]: Briefly state where you can get these.

File: examples/notebooks/datastores/7_cmems_data_store.ipynb [hunk 3]
```diff
@@ -0,0 +1,2538 @@
...
+   "source": [
+    "JSON(list(store.get_data_ids()))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4df9dd30-c230-4bdb-8d5d-725deeb36753",
+   "metadata": {},
+   "source": [
+    "#### Get more info about a specific dataset. This includes a description of the possible open formats:"
```
Comment (TonioF) [inline examples/notebooks/datastores/7_cmems_data_store.ipynb:758]:
```suggestion
    "#### Get more info about a specific dataset."
```
I realize you just copied this from other notebooks, but I don't know what is meant with "open formats" here.

File: examples/notebooks/datastores/7_cmems_data_store.ipynb [hunk 4]
```diff
@@ -0,0 +1,2538 @@
...
+   "source": [
+    "store.get_open_data_params_schema('dataset-bal-analysis-forecast-wav-hourly')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "937fef03-6e92-4cd0-86cd-b4d0f0ebbd9e",
+   "metadata": {},
+   "source": [
+    "#### The dataset can be opened by giving variables_names, time_range and bbox as params. But note that none of them are mandatory"
```
Comment (TonioF) [inline examples/notebooks/datastores/7_cmems_data_store.ipynb:1308]:
```suggestion
    "#### The dataset can be opened by giving variables_names, time_range and bbox as params, but none of them are mandatory."
```

**Review by TonioF — CHANGES_REQUESTED** [review]
- Can you please run the notebooks so the cell outputs are updated?
- Please add a heading to the cmems notebook
- Have a look at my other comments

**Review by forman — CHANGES_REQUESTED** [review]
Please adhere _exactly_ to the style of the other Notebooks, e.g.,

It is not "XCUBE", but "xcube":

image

Do not use bold header texts:

image

**Comment by TonioF** [comment]
I'm getting confused. Why are there now changes to other modules?

**Comment by TejasMorbagal** [comment]
> I'm getting confused. Why are there now changes to other modules?

Because I merged master into this branch, as Norman asked me.

**Comment by TonioF** [comment]
> > I'm getting confused. Why are there now changes to other modules?
> 
> Because I merged master into this branch, as Norman asked me.

I see. It's weird github lists this as a changed file though.

File: examples/notebooks/datastores/7_cmems_data_store.ipynb [hunk 5]
```diff
@@ -0,0 +1,2547 @@
...
+   "source": [
+    "store.describe_data('dataset-bal-analysis-forecast-wav-hourly')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e2f3e393-87e4-4a62-bd4a-0b7d1e6df9fb",
+   "metadata": {},
+   "source": [
+    "Which parameters must be passsed or are available to open the dataset? "
```
Comment (TonioF) [inline examples/notebooks/datastores/7_cmems_data_store.ipynb:1213]:
```suggestion
    "Which parameters must be passed or are available to open the dataset? "
```

**Review by TonioF — CHANGES_REQUESTED** [review]
Found one more little thing (see comment). Also, unfortunately I was unable to run the notebook because of a NotJSONError. Can you please have a look?

**Comment by TejasMorbagal** [comment]
> Found one more little thing (see comment). Also, unfortunately I was unable to run the notebook because of a NotJSONError. Can you please have a look?

Notebook runs fine for me. Have you set cmems credentials as ENV VAR?

**Comment by TonioF** [comment]
> > Found one more little thing (see comment). Also, unfortunately I was unable to run the notebook because of a NotJSONError. Can you please have a look?
> 
> Notebook runs fine for me. Have you set cmems credentials as ENV VAR?

I don't even get there. I receive this error message when I try to open the notebook: 
image

**Comment by pont-us** [comment]
> I don't even get there. I receive this error message when I try to open the notebook:

@TonioF Same here. Looks as though commit 241c196b5 broke it -- full stop character outside the string on line 154.

**Comment by TejasMorbagal** [comment]
> > I don't even get there. I receive this error message when I try to open the notebook:
> 
> @TonioF Same here. Looks as though commit [241c196](https://github.com/dcs4cop/xcube/commit/241c196b5cfa37deb488b63683e2d990ea61d52f) broke it -- full stop character outside the string on line 158.

@pont-us @TonioF Yes, I reverted the commit. It works now. Kindly check and confirm

File: examples/notebooks/datastores/7_cmems_data_store.ipynb [hunk 6]
```diff
@@ -0,0 +1,2512 @@
...
+   "source": [
+    "get_data_store_params_schema('cmems')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "214307bc-7def-433f-876e-28540a6c2fe7",
+   "metadata": {},
+   "source": [
+    "set `CMEMS_USER` and `CMEMS_PASSWORD` as environment variables\n",
```
Comment (AliceBalfanz) [inline examples/notebooks/datastores/7_cmems_data_store.ipynb:153]:
```suggestion
    "Set `CMEMS_USER` and `CMEMS_PASSWORD` as environment variables\n",
```

File: examples/notebooks/datastores/7_cmems_data_store.ipynb [hunk 7]
```diff
@@ -0,0 +1,2512 @@
...
+   "source": [
+    "store.get_open_data_params_schema('dataset-bal-analysis-forecast-wav-hourly')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "937fef03-6e92-4cd0-86cd-b4d0f0ebbd9e",
+   "metadata": {},
+   "source": [
+    "The dataset can be opened by giving variables_names, time_range and bbox as params, but none of them are mandatory."
```
Comment (AliceBalfanz) [inline examples/notebooks/datastores/7_cmems_data_store.ipynb:1280]:
```suggestion
    "The dataset can be opened by giving `variables_names`, `time_range` and `bbox` as params, but none of them are mandatory."
```

**Review by AliceBalfanz — APPROVED** [review]
Minor change requests, but I don't insist :)

**Review by forman — APPROVED** [review]

**Pull request merged**